### PR TITLE
Loading states uneditable 2

### PR DESCRIPTION
**Pull request by Marc-Andre-Rivet**
Fixing the test issues we've been encountering with #484 

Modifies the tests so as to take into account whether the table is ready/loading/any as the test runs. Some actions require specific states to be reached while others do not.

Also reworks the copy/paste test server and the loading test servers to maximize the use of `preventupdate` for null-change scenarios, speeding up the tests significantly.

Also adds a "default" test server for tests that were using the copy/paste server but didn't need to, making them simpler (or at least preventing them from needing additional changes).

Proposing these fixes for merge in the feature PR #484

### CODE REVIEW

File: src/dash-table/components/ControlledTable/index.tsx [hunk 1]
```diff
@@ -766,7 +769,8 @@ export default class ControlledTable extends PureComponent<ControlledTableProps>
             ...(visibleColumns.length ? [] : ['dash-no-columns']),
             ...(virtualized.data.length ? [] : ['dash-no-data']),
             ...(filter_action !== TableAction.None ? [] : ['dash-no-filter']),
-            ...(fill_width ? ['dash-fill-width'] : [])
+            ...(fill_width ? ['dash-fill-width'] : []),
+            ...(isLoading ? ['dash-loading'] : [])
```
Comment (Marc-Andre-Rivet) [inline src/dash-table/components/ControlledTable/index.tsx]: Additional class while loading

File: tests/cypress/dash/v_copy_paste.py [hunk 2]
```diff
@@ -85,15 +86,20 @@
...
     if len(current) == len(previous):
         for (i, datum) in enumerate(current):
             previous_datum = previous[i]
             if datum[0] != previous_datum[0]:
+                modified = True
                 datum[1] = "MODIFIED"
 
+    if not modified:
+        raise PreventUpdate
+
```
Comment (Marc-Andre-Rivet) [inline tests/cypress/dash/v_copy_paste.py]: Prevent update when there's nothing to update

File: tests/cypress/dash/v_data_loading.py [hunk 3]
```diff
@@ -59,24 +61,27 @@
...
-def triggerWait(to_change, current):
-    if to_change == 'change_data':
-        sleep(5)
-    return current
+def triggerWait(to_change):
+    if to_change != 'change_data':
+        raise PreventUpdate
+
+    sleep(5)
+    return df[0:250]
```
Comment (Marc-Andre-Rivet) [inline tests/cypress/dash/v_data_loading.py]: Remove the need for `data` as state by returning the baseline. copy/paste already tests that state/output data works

File: tests/cypress/src/DashTable.ts [hunk 4]
```diff
@@ -1,26 +1,47 @@
...
+        default:
+            return ANY;
+    }
+};
+
 export default class DashTable {
-    static getCell(row: number, column: number) {
-        return cy.get(`#table tbody tr td.column-${column}`).eq(row);
+    static getCell(row: number, column: number, editable: State = State.Editable) {
+        return cy.get(`#table ${getSelector(editable)} tbody tr td.column-${column}`).eq(row);
```
Comment (Marc-Andre-Rivet) [inline tests/cypress/src/DashTable.ts]: With the loading state that affects the table's rendering and behavior it's now important to be able to select the cells in the correct state (or any state) based on what's tested. Default is `editable` for backward compatibility with the tests.

File: tests/cypress/tests/server/copy_paste_test.ts [hunk 5]
```diff
@@ -1,10 +1,11 @@
-import DashTable from 'cypress/DashTable';
+import DashTable, { State } from 'cypress/DashTable';
 import DOM from 'cypress/DOM';
 import Key from 'cypress/Key';
 
 describe('copy paste', () => {
     beforeEach(() => {
         cy.visit('http://localhost:8082');
+        cy.wait(1000);
```
Comment (Marc-Andre-Rivet) [inline tests/cypress/tests/server/copy_paste_test.ts]: There's a double loading loop for data as input and state. Give it some time just in case.

File: tests/cypress/tests/server/copy_paste_test.ts [hunk 6]
```diff
@@ -14,19 +15,11 @@ describe('copy paste', () => {
...
-	 * in which a cell is being edited.
-	 */
-        DashTable.getCell(3, 0).get('input');
         DOM.focused.type(`${Key.Meta}v`);
-        DashTable.getCell(0, 0).click();
+        DashTable.getCell(0, 0, State.Any).click();
 
         for (let row = 0; row <= 2; ++row) {
-            DashTable.getCell(row + 3, 0).within(() => cy.get('.dash-cell-value').should('have.html', `${row}`));
+            DashTable.getCell(row + 3, 0, State.Any).within(() => cy.get('.dash-cell-value').should('have.html', `${row}`));
```
Comment (Marc-Andre-Rivet) [inline tests/cypress/tests/server/copy_paste_test.ts]: This test does not require the table to be ready or loading, so it just gets whatever there is.

File: tests/cypress/tests/server/loading_uneditable_test.ts [hunk 7]
```diff
@@ -9,173 +9,159 @@ describe('loading states uneditable', () => {
     });
 
     it('does not permit editing when data are loading', () => {
-
+        // Table is editable
```
Comment (Marc-Andre-Rivet) [inline tests/cypress/tests/server/loading_uneditable_test.ts]: Simplified these tests somewhat but I think I'm staying true to the intent.
